values/dict.go

```diff
@@ -232,7 +232,7 @@ func dictComparer(dictType semantic.MonoType) immutable.Comparer {
 	case semantic.Time:
 		return timeComparer{}
 	default:
-		panic(errors.Newf(codes.Internal, "invalid key nature: %s", n))
+		return intComparer{}
```

Comment:
Whats the reasoning behind this change?

Reply:
Thanks for catching this. I had meant to revert this.